.github/actions/conventional-checks/action.yml

```diff
@@ -0,0 +1,49 @@
+name: Conventional Checks
+
+description: Check if the branch name and pull request title follow conventional checks.
+
+runs:
+  using: composite
+  steps:
+    - name: Get branch name (merge)
+      if: github.event_name != 'pull_request'
+      shell: bash
+      run: echo "BRANCH_NAME=$(echo ${GITHUB_REF#refs/heads/})" >> $GITHUB_ENV
+
+    - name: Get branch name (pull request)
+      if: github.event_name == 'pull_request'
+      shell: bash
+      run: echo "BRANCH_NAME=$(echo ${GITHUB_HEAD_REF})" >> $GITHUB_ENV
```

Comment:
we already have a `get-branch-name` action which does exactly this :)

you can embed it here or add a mandatory input for branch name in case it may need further customization

Reply:
Done